tests/tests/tx/upgrade.rs

```diff
@@ -168,6 +169,115 @@ async fn can_upgrade_to_uploaded_state_transition() {
+    let mut test_builder = TestSetupBuilder::new(2322);
+    test_builder.utxo_validation = false;
+    test_builder.privileged_address = privileged_address;
+    let TestContext {
+        client,
+        srv: _drop,
+        mut rng,
+        ..
+    } = test_builder.finalize().await;
+
+    let transactions = transactions_from_subsections(&mut rng, subsections, amount);
+    for upload in transactions {
+        let mut tx = upload.into();
+        client.estimate_predicates(&mut tx).await.unwrap();
+        client.submit_and_await_commit(&tx).await.unwrap();
+    }
+
+    // Given
+    let upgrade = Transaction::upgrade(
+        UpgradePurpose::StateTransition { root },
+        Policies::new().with_max_fee(amount),
+        vec![Input::coin_predicate(
+            rng.gen(),
+            privileged_address,
+            amount,
+            AssetId::BASE,
+            Default::default(),
+            Default::default(),
+            predicate(),
+            vec![],
+        )],
+        vec![],
+        vec![],
+    );
+
+    // When
+    let mut tx = upgrade.into();
+    client.estimate_predicates(&mut tx).await.unwrap();
+    let result = client.submit_and_await_commit(&tx).await;
+
+    // Then
+    let result_str = format!("{:?}", result); // io::Result forces string handling
+    result.expect_err("Upgrading to an incorrect bytecode should fail");
+    assert!(
+        result_str.contains("WASM bytecode contents are not valid"),
+        "msg: {}",
+        result_str
+    );
+}
+
+#[tokio::test]
+async fn upgrading_to_missing_state_transition_fails() {
+    let privileged_address = Input::predicate_owner(predicate());
+    let amount = 1_000;
+    let mut test_builder = TestSetupBuilder::new(2322);
+    test_builder.utxo_validation = false;
+    test_builder.privileged_address = privileged_address;
+    let TestContext {
+        client,
+        srv: _drop,
+        mut rng,
+        ..
+    } = test_builder.finalize().await;
+
+    // Given
+    let upgrade = Transaction::upgrade(
+        UpgradePurpose::StateTransition {
+            root: Bytes32::new([1; 32]),
+        },
+        Policies::new().with_max_fee(amount),
+        vec![Input::coin_predicate(
+            rng.gen(),
+            privileged_address,
+            amount,
+            AssetId::BASE,
+            Default::default(),
+            Default::default(),
+            predicate(),
+            vec![],
+        )],
+        vec![],
+        vec![],
+    );
+
+    // When
+    let mut tx = upgrade.into();
+    client.estimate_predicates(&mut tx).await.unwrap();
+    let result = client.submit_and_await_commit(&tx).await;
+
+    // Then
+    let result_str = format!("{:?}", result); // io::Result forces string handling
+    result.expect_err("Upgrading to missing bytecode should fail");
+    assert!(
+        result_str.contains("WASM bytecode matching the given root was not found"),
+        "msg: {}",
+        result_str
+    );
+}
+
```

Comment:
Maybe we want to test the case with incomplete bytecode

Reply:
71744b63c